server/server.go

```diff
@@ -829,6 +829,15 @@ func (server *Server) loadConfig(configurations configs, globalConfiguration Glo
 						}
 						negroni.Use(cbreaker)
 					} else {
+						if frontend.Headers.HasCustomHeadersDefined() {
```

Comment:
That is an excellent point. I agree, I will move it outside. Somewhere along my testing, I found that location worked, but on review, I agree, moving out of that block is ideal.